spec/controllers/api/v1/application_users_controller_spec.rb

```diff
@@ -46,6 +45,34 @@
     end
   end
 
+  describe "find by username returns" do
+    it "a single result when username matches" do
+      api_get :find_by_username, untrusted_application_token, parameters: { username: 'foo_bb' }
+      expect(response.code).to eq('200')
+      expected_response = [{
+        id: bob_brown.application_users.first.id,
+        user: {
+          id: bob_brown.id,
+          username: bob_brown.username,
+          first_name: bob_brown.first_name,
+          last_name: bob_brown.last_name
+        }, unread_updates: 0
+      }].to_json
+      expect(response.body).to eq(expected_response)
+    end
+    it "raises not found when when not found" do
+      expect {
+        api_get :find_by_username, untrusted_application_token, parameters: { username: 'foo' }
+      }.to raise_error(ActiveRecord::RecordNotFound)
+    end
+    it "only finds users belonging to the requesting application" do
+      expect {
+        api_get :find_by_username, trusted_application_token, parameters: { username: 'billy_20' }
```

Comment:
Actually it does, it's created as part of the "billy_users" block at the top of the spec.  I just picked the 20th at random.  https://github.com/openstax/accounts/blob/application_users-username-search/spec/controllers/api/v1/application_users_controller_spec.rb#L23-L33

I'm re-working the spec to be more explicit though since it's non-obvious what it's testing :)